util/include/crypto/eddsa/EdDSASigner.hpp

```diff
@@ -34,22 +56,25 @@ class EdDSASigner {
   }
 
   void sign(const uint8_t *msg, size_t len, uint8_t *signature, size_t &signatureLength) const {
-    concord::util::openssl_utils::UniquePKEY pkey{EVP_PKEY_new_raw_private_key(
-        NID_ED25519, nullptr, privateKey_.getBytes().data(), privateKey_.getBytes().size())};
     concord::util::openssl_utils::UniqueOpenSSLContext ctx{EVP_MD_CTX_new()};
-    ConcordAssertEQ(EVP_DigestSignInit(ctx.get(), nullptr, nullptr, nullptr, pkey.get()),
-                    concord::util::openssl_utils::OPENSSL_SUCCESS);
-    ConcordAssertEQ(EVP_DigestSign(ctx.get(), signature, &signatureLength, msg, len),
-                    concord::util::openssl_utils::OPENSSL_SUCCESS);
+    ConcordAssertEQ(EVP_DigestSignInit(ctx.get(), nullptr, nullptr, nullptr, pkey_.get()), OPENSSL_SUCCESS);
+    ConcordAssertEQ(EVP_DigestSign(ctx.get(), signature, &signatureLength, msg, len), OPENSSL_SUCCESS);
   }
 
-  std::string sign(const std::string &message) const {
+  std::string sign(const std::string &message) override {
     return sign(reinterpret_cast<const uint8_t *>(message.data()), message.size());
   }
 
-  const PrivateKeyType &getPrivKey() const { return privateKey_; }
+  uint32_t signatureLength() const override { return EdDSASignatureByteSize; }
+
+  std::string getPrivKey() const override { return privateKey_.toString(); }
+
+  const PrivateKeyType &getPrivateKey() const { return privateKey_; }
```

Comment:
`EdDSAMultisigSigner::getShareSecretKey()` calls this. It expects `IShareSecretKey` type.

Reply:
These functions should be implemented by EdDSAMultisigSigner. not by EdDSASigner.

Reply:
I couldn't find and usages for getPrivKey.
Since it is just a duplication of toString, I suggest removing it.
privateKey_ can be made protected and shared with the threshsign class this way.

Reply:
Changes in https://github.com/arc-vmware/concord-bft/pull/6/commits/87f4edd0f4f5d72b422cef50f4a8eeeee8b28bc2